### PR TITLE
Rekey identification with fix

### PR DESCRIPTION
**Pull request by taruti**

### CODE REVIEW

**Comment by taruti** [comment]
This fixes the flake encountered after the merge.
1) Revert the revert
2) Handle the case where we rekey and there was a logic race (the other code paths are ok)
3) Will be merged after Jenkins passes three times as to avoid flakes

**Comment by taruti** [comment]
3 runs of Jenkins have passed green.

**Comment by taruti** [comment]
Appveyor not related.

File: libkbfs/folder_branch_ops.go [hunk 1]
```diff
@@ -1053,6 +1053,11 @@ func (fbo *folderBranchOps) getMDForWriteOrRekeyLocked(
 	headStatus := headTrusted
 	if mdType == mdRekey {
 		headStatus = headUntrusted
+		// If we already have a head (that has been filled  after the initial
```
Comment (strib) [inline libkbfs/folder_branch_ops.go]: Extra space after "filled".

File: libkbfs/folder_branch_ops.go [hunk 2]
```diff
@@ -1053,6 +1053,11 @@ func (fbo *folderBranchOps) getMDForWriteOrRekeyLocked(
 	headStatus := headTrusted
 	if mdType == mdRekey {
 		headStatus = headUntrusted
```
Comment (strib) [inline libkbfs/folder_branch_ops.go]: This could be moved below the `if` block to save an instruction.

File: libkbfs/folder_branch_ops.go [hunk 3]
```diff
@@ -973,7 +1050,16 @@ func (fbo *folderBranchOps) getMDLocked(
 
 	fbo.headLock.Lock(lState)
 	defer fbo.headLock.Unlock(lState)
-	err = fbo.setInitialHeadUntrustedLocked(ctx, lState, md)
+	headStatus := headTrusted
+	if mdType == mdRekey {
+		headStatus = headUntrusted
```
Comment (strib) [inline libkbfs/folder_branch_ops.go]: This could be moved below the if block to save an instruction.

File: libkbfs/folder_branch_ops.go [hunk 4]
```diff
@@ -973,7 +1050,16 @@ func (fbo *folderBranchOps) getMDLocked(
...
+		// If we already have a head (that has been filled  after the initial
```
Comment (strib) [inline libkbfs/folder_branch_ops.go]: Extra space after "filled".

**Pull request merged**